### PR TITLE
Post Temporary Appointment During Calendar Form Submission 

### PR DESCRIPTION
**Pull request by will0684**
When a user submits the calendar page form (i.e. when they hit "Next"), the app will attempt to communicate with the server to create a temporary appointment (with database TTL of 5 minutes). If there is an error, they are redirected to the error page. If it succeeds, the response (the temporary appointment) is saved to the store under the calendar property. The object ID will be used to confirm the appointment in thee API call.

### CODE REVIEW

File: src/pages/CalendarPage.js [hunk 1]
```diff
@@ -168,6 +171,29 @@ class CalendarPage extends Component {
...
+    let appointment = {
+      clientEmail: values.register.email,
+      locationId: values.selectProvince.locationId,
+      // TODO: Get bioKit information during selection of timeslot
+      // bioKitId: String,
+      bil: values.register.paperFileNumber,
+      date: date,
+      privateAccessible: false,
+    }
+    // console.log(appointment)
```
Comment (andr0272) [inline src/pages/CalendarPage.js]: y tho :thinking:

File: src/pages/CalendarPage.js [hunk 2]
```diff
@@ -168,6 +171,29 @@ class CalendarPage extends Component {
...
+    return axios.post(`/appointments/temp`, appointment)
+  }
+
   async onSubmit(values, event) {
     // values.selectedDays (when set) is an array of dates, so cast values to ISO date strings
```
Comment (andr0272) [inline src/pages/CalendarPage.js]: y tho :thinking:

File: src/server.js [hunk 3]
```diff
@@ -41,6 +40,7 @@ server
   .use(ensureLocation)
   .use(cookieParser())
   .use(bodyParser.urlencoded({ extended: false }))
+  .use(bodyParser.json())
```
Comment (andr0272) [inline src/server.js]: Is it possible to chain these?

File: src/pages/CalendarPage.js (hunk 2 above)
Comment (newcitywave) [reply src/pages/CalendarPage.js]: why though!

**Comment by will0684** [comment]
@andr0272 You have to check how many total biokits the location has and create as many appointments as there are biokits to block out a section right now. We're still working on the accessibility/privacy toggle being passed as well

File: src/pages/CalendarPage.js (hunk 1 above)
Comment (will0684) [reply src/pages/CalendarPage.js]: Deleted the comment line

**Pull request merged**